core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -538,6 +548,13 @@ public void failEngine(String reason, Throwable failure) {
                     logger.warn("failed engine [{}]", failure, reason);
                     // we must set a failure exception, generate one if not supplied
                     failedEngine = failure;
+                    if (markAsCorrupted && markedAsCorrupted == false) {
```

Comment:
what what ? it took me 10s seconds to seed the ed in marked but also, why do need to do it twice? can't we just always do it in the finally clause?